File: src.csharp/AlphaTab/Core/Dom/TextDecoder.cs
using System.Text;
using AlphaTab.Core.EcmaScript;

namespace AlphaTab.Core.Dom;

internal class TextDecoder
{
    private readonly Encoding _encoding;

    public TextDecoder(string encoding)
    {
        _encoding = Encoding.GetEncoding(encoding);
    }

    public string Decode(ArrayBuffer data)
    {
        return _encoding.GetString(data.Raw, 0, (int)data.ByteLength);
    }
}

File: packages/kotlin/android/src/main/java/alphaTab/core/dom/TextDecoder.kt
package alphaTab.core.dom

import alphaTab.core.decodeToString
import alphaTab.core.ecmaScript.ArrayBuffer

internal class TextDecoder(encoding:String) {
    private val _encoding:String = encoding

    @ExperimentalUnsignedTypes
    public fun decode(buffer: ArrayBuffer): String {
        return buffer.decodeToString(_encoding)
    }
}
